src/meta/src/hummock/manager/mod.rs

```diff
@@ -1450,14 +1448,20 @@ where
             }
         }
 
-        if need_cancel_task {
-            if let Err(e) = self.cancel_compact_task(&mut compact_task).await {
-                tracing::error!("Failed to cancel task {}. {:#?}", compact_task.task_id, e);
-                // TODO #3677: handle cancellation via compaction heartbeat after #4496
-            }
-        }
-
         if is_failed {
+            let hummock_manager = self.clone();
+            tokio::spawn(async move {
+                while let Err(e) = hummock_manager.cancel_compact_task(&mut compact_task).await {
```

Comment:
I'm not sure we should do this as it can lead to CPU leak...

Maybe a fixed number of retries?

Reply:
Refactored. Extend the existed hummock worker, to handle cancellation request sequentially.